packages/devtools_app/lib/src/inspector/inspector_controller.dart

```diff
@@ -133,6 +133,34 @@ class InspectorController extends DisposableController
         }
       });
     }
+
+    autoDispose(
+      serviceManager.onConnectionAvailable.listen(_handleConnectionStart),
+    );
+    if (serviceManager.connectedAppInitialized) {
+      _handleConnectionStart(serviceManager.service);
+    }
+    autoDispose(
+      serviceManager.onConnectionClosed.listen(_handleConnectionStop),
+    );
+  }
+
+  void _handleConnectionStart(VmService service) {
+    // Clear any existing badge/errors for older errors that were collected.
+    // Do this in a post frame callback so that we are not trying to clear the
+    // error notifiers for this screen while the framework is already in the
+    // process of building widgets.
+    WidgetsBinding.instance.addPostFrameCallback((_) {
```

Comment:
can you remove this addPostFrameCallback now that this is in the controller? I would have thought its lifecycle was different from State objects so this wouldn't be needed.

Reply:
nm. maybe add a TODO that when the inspectorController isn't called from 
createState() this can be removed.

Reply:
We still need this in a post frame callback because it triggers a value listenable builder elsewhere in the widget tree that is already building. Added TODO in https://github.com/flutter/devtools/pull/3347